packages/blocks/src/__internal__/rich-text/rich-text.ts

```diff
@@ -136,6 +136,29 @@ export class RichText extends ShadowlessElement {
 
           ctx.attributes = attributes ?? null;
         }
+
+        if (vRange.index >= 0) {
+          const selection = window.getSelection();
+          if (selection && selection.rangeCount !== 0) {
+            const range = selection.getRangeAt(0);
+            const container = range.startContainer;
+
+            if (
+              container instanceof Text &&
+              container.parentElement?.dataset.virgoText === 'true'
+            ) {
+              const [text] = vEditor.getTextPoint(vRange.index);
+              const affineReference =
+                text.parentElement?.closest('affine-reference');
+              if (
+                affineReference &&
+                text.textContent !== ZERO_WIDTH_NON_JOINER
+              ) {
+                text.textContent = ZERO_WIDTH_NON_JOINER;
+              }
+            }
```

Comment:
I do not advocate special treatment for use cases, as this does not completely solve the problem and makes the `rich-text` difficult to maintain.

@Flrande Do you have any other effective approaches for this situation?

Reply:
I understand your concerns just as I comment. So maybe we could find a more universal approach.
I consider some way last night:
1. We add the custom ime handler to override in handler context https://github.com/toeverything/blocksuite/blob/master/packages/virgo/src/services/event.ts#L11
```ts
export interface VHandlerContext<
  T extends BaseTextAttributes,
  E extends Event = Event
> {
  event: E;
  data: string | null;
  vRange: VRange;
  skipDefault: boolean;
  attributes: T | null;
  imeHandler: () => boolean; // boolean represent whether we need to use the default way to solve
}
```
But it may pollute other handler since it may only occured in compositionend.

2. We add the custom ime handler to as part of the virgo handler we bind. https://github.com/toeverything/blocksuite/blob/master/packages/virgo/src/services/event.ts#L30
```ts
private _handlers: {
    keydown?: (event: KeyboardEvent) => void;
    paste?: (event: ClipboardEvent) => void;
    // corresponding to native input event and used to take over default behavior in virgo
    virgoInput?: (
      ctx: VHandlerContext<TextAttributes, InputEvent>
    ) => VHandlerContext<TextAttributes, InputEvent>;
    // corresponding to native compositionend event and used to take over default behavior in virgo
    virgoCompositionEnd?: (
      ctx: VHandlerContext<TextAttributes, CompositionEvent>
    ) => VHandlerContext<TextAttributes, CompositionEvent>;
    imeHandler: () => boolean;
  } = {};
```
But I think imeHandler is contained by virgoCompositionEnd, and it is not the same level with other handler.
3. Maybe we can find a way to make reference node implementation use v-text as well https://github.com/toeverything/blocksuite/blob/master/packages/blocks/src/__internal__/rich-text/reference-node.ts#L176. And it may also recommend any other custom render use v-text to make the logic more consistent.

Reply:
> I do not advocate special treatment for use cases, as this does not completely solve the problem and makes the rich-text difficult to maintain.
> 
> @Flrande Do you have any other effective approaches for this situation?

Before we implement a better abstraction of Virgo's exposed interface, I think we can make this fix take effect.